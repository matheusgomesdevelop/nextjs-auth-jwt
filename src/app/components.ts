// Atoms
import {
  AAvatar,
  ATitle,
  A_Label,
  A_Input,
  ACheckbox,
  A_Button,
} from "@/components/atoms";

// Molecules
import { M_InputWithLabel } from "@/components/molecules";

export {
  AAvatar,
  ATitle,
  A_Label,
  A_Input,
  ACheckbox,
  A_Button,
  M_InputWithLabel,
};
