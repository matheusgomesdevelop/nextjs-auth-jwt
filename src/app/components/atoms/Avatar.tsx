import React from 'react';

import * as PrAvatar from 'primereact/avatar';

interface AvatarProps extends PrAvatar.AvatarProps {}

const Avatar: React.FC<AvatarProps> = ({ image }) => (
    <PrAvatar.Avatar
        className="w-32 h-32"
        image={image}
        pt={{
            image: {
                className: 'rounded-[32px]',
            },
        }}
    />
);

export default Avatar;
