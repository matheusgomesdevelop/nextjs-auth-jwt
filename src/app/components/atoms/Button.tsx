import React, { forwardRef, useImperativeHandle, useState } from 'react';

import * as PrButton from 'primereact/button';

interface ButtonProps extends PrButton.ButtonProps {
    'data-testid'?: string;
    variant: 'fwMd-fs16-colGray700-bgWhite' | 'gradient' | 'transparent';
}

export interface ButtonRef {
    setLoading: React.Dispatch<React.SetStateAction<boolean>>;
}

const Button: React.ForwardRefRenderFunction<ButtonRef, ButtonProps> = ({ variant, children, className }, ref) => {
    const [loading, setLoading] = useState(false);

    useImperativeHandle(ref, () => ({ setLoading }), []);

    return (
        <PrButton.Button
            loading={loading}
            className={`px-[10px] py-[14px] h-[44px] w-full rounded-8 border-1 border-solid hover:opacity-90 transition duration-150 ${
                variant === 'transparent'
                    ? 'border-none bg-transparent text-gray-900'
                    : variant === 'fwMd-fs16-colGray700-bgWhite'
                      ? 'text-md text-gray-700 bg-white border-gray-100 hover:border-primary'
                      : variant === 'gradient'
                        ? 'bg-gradient-to-r from-primary to-secondary font-semibold text-white border-secondary'
                        : ''
            } 
   ${className} `}
        >
            {!loading && children}
        </PrButton.Button>
    );
};

export default forwardRef(Button);
