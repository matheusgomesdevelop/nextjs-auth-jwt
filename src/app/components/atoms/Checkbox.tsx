import React from 'react';

import * as PrCheckbox from 'primereact/checkbox';
import { classNames } from 'primereact/utils';

export interface CheckboxProps extends PrCheckbox.CheckboxProps {
    checked: boolean;
}

const Checkbox: React.FC<CheckboxProps> = ({ checked, ...props }) => (
    <PrCheckbox.Checkbox
        className="appearance-none checked:bg-secondary"
        pt={{
            root: {
                className: classNames('cursor-pointer inline-flex relative select-none align-bottom', 'w-6 h-6'),
            },
            input: {
                className: classNames(
                    'absolute appearance-none top-0 left-0 size-full p-0 m-0 opacity-0 z-10 outline-none cursor-pointer'
                ),
                style: {
                    minWidth: '16px',
                    minHeight: '16px',
                },
            },
            box: ({ props, context }: PrCheckbox.CheckboxPassThroughMethodOptions) => ({
                className: classNames(
                    'flex items-center justify-center',
                    'border-2 w-6 h-6 text-gray-600 rounded-sm transition-colors duration-200',
                    {
                        'border-gray-300 bg-white dark:border-blue-900/40 dark:bg-gray-900': !context.checked,
                        'border-blue-500 bg-blue-500 dark:border-blue-400 dark:bg-blue-400': context.checked,
                    },
                    {
                        'hover:border-blue-500 dark:hover:border-blue-400 focus:outline-none focus:outline-offset-0 focus:shadow-[0_0_0_0.2rem_rgba(191,219,254,1)] dark:focus:shadow-[inset_0_0_0_0.2rem_rgba(147,197,253,0.5)]':
                            !props.disabled,
                        'cursor-default opacity-60': props.disabled,
                    }
                ),

                style: {
                    minWidth: '16px',
                    minHeight: '16px',
                },
            }),
            icon: {
                className: 'w-4 h-4 transition-all duration-200 text-white text-base dark:text-white',
                style: {
                    width: '12px',
                    height: '12px',
                },
            },
        }}
        checked={checked}
        {...props}
    />
);

export default Checkbox;
