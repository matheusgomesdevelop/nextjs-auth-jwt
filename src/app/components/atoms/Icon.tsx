import React from 'react';

interface IconProps {
    icon: 'pi-lock' | 'pi-github' | 'pi-linkedin' | 'pi-sun' | 'pi-spin pi-cog';
}

const Icon: React.FC<IconProps> = ({ icon }) => <i className={`text-white text-lg pi ${icon}`} />;

export default Icon;
