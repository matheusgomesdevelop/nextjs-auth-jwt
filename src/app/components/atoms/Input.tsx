import React from 'react';

import { InputTextProps, InputText } from 'primereact/inputtext';

export interface InputProps extends InputTextProps {}

const Input: React.FC<InputProps> = ({ ...props }) => <InputText {...props} />;

export default Input;
