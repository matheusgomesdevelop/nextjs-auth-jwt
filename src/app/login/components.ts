import { Avatar } from "@/components/atoms";

export { Avatar };
