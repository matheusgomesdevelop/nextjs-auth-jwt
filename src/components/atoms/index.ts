import Checkbox from './Checkbox/Checkbox';
import Button, { ButtonRef } from './Button/Button';
import Avatar from './Avatar/Avatar';
import Label from './Label/Label';
import Input from './Input/Input';
import Title from './Title/Title';
import Text from './Text/Text';
import Icon from './Icon/Icon';
import Link from './Link/Link';

export { Checkbox, Button, Avatar, Title, Label, Input, Text, Icon, Link };

export type { ButtonRef };
