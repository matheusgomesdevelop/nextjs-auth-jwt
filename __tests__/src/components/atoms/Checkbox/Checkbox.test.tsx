import React from 'react';

import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';

import { Checkbox } from '@/components/atoms';

describe('Deve renderizar o Checkbox corretamente', () => {
    it('Deve renderizar o Checkbox na tela', () => {
        const getCheckbox = renderer.create(<Checkbox checked={false} data-testid="checkbox-testid" />).toJSON();

        expect(getCheckbox).toMatchSnapshot();
    });
});
