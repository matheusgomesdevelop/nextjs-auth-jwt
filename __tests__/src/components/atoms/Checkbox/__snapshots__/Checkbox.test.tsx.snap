// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o Checkbox corretamente Deve renderizar o Checkbox na tela 1`] = `
<div
  className="p-checkbox p-component"
  data-pc-name="checkbox"
  data-pc-section="root"
  data-testid="checkbox-testid"
  id={null}
  onClick={[Function]}
  onContextMenu={null}
  onMouseDown={null}
  style={{}}
>
  <div
    className="p-hidden-accessible"
    data-pc-section="hiddeninputwrapper"
  >
    <input
      data-pc-section="hiddeninput"
      defaultChecked={false}
      disabled={false}
      id={null}
      name={null}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      readOnly={false}
      required={false}
      tabIndex={null}
      type="checkbox"
    />
  </div>
  <div
    className="p-checkbox-box"
    data-p-disabled={false}
    data-p-focus={false}
    data-p-highlight={false}
    data-pc-section="input"
  />
</div>
`;
